check_run/check_run/report/ach_prenote/ach_prenote.py

```diff
@@ -0,0 +1,295 @@
+		},
+		{
+			"label": frappe._("Party"),
+			"fieldname": "party",
+			"fieldtype": "Data",
+			"width": "300px",
+			"hidden": True,
+		},
+		{
+			"label": frappe._("Party Name"),
+			"fieldname": "party_name",
+			"fieldtype": "Data",
+			"width": "300px",
+		},
+		{
+			"label": frappe._("Bank"),
+			"fieldname": "bank",
+			"fieldtype": "Link",
+			"width": "200px",
+			"options": "Bank",
+		},
+		{
+			"label": frappe._("Account Type"),
+			"fieldname": "ach_account_type",
+			"fieldtype": "Data",
+			"width": "150px",
+		},
+		{
+			"label": frappe._("ACH Prenote Date"),
+			"fieldname": "ach_prenote_date",
+			"fieldtype": "Date",
+			"width": "150px",
+		},
+		{
+			"label": frappe._("Last Used Date"),
+			"fieldname": "ach_last_used",
+			"fieldtype": "Date",
+			"width": "150px",
+		},
+		{
+			"label": frappe._("Validated Date"),
+			"fieldname": "account_details_validated",
+			"fieldtype": "Date",
+			"width": "200px",
+			"editable": True,
+		},
+	]
+
+
+def get_data(filters):
+	Supplier = frappe.qb.DocType("Supplier")
+	Employee = frappe.qb.DocType("Employee")
+	supplier_query = (
+		frappe.qb.from_(Supplier)
+		.select(
+			Supplier.ach_last_used,
+			Supplier.account_details_validated,
+			Supplier.bank,
+			Supplier.ach_account_type,
+			Supplier.ach_prenote_date,
+			(Supplier.supplier_name).as_("party_name"),
+			(Supplier.name).as_("party"),
+			ConstantColumn("Supplier").as_("party_type"),
+		)
+		.where(Supplier.bank.isnotnull())
+		.where(Coalesce(Supplier.ach_last_used, "1900-1-1") <= getdate(filters.last_used_date))
+		.where(
+			Coalesce(Supplier.account_details_validated, "1900-1-1") <= getdate(filters.validated_date)
+		)
+		.orderby(
+			Supplier.ach_last_used,
+			Supplier.account_details_validated,
+		)
+	)
+	if filters.ach_account_type == "Checking":
+		supplier_query.where(Supplier.ach_account_type == "Checking")
+	elif filters.ach_account_type == "Savings":
+		supplier_query.where(Supplier.ach_account_type == "Savings")
+	suppliers = supplier_query.run(as_dict=True)
+	employee_query = (
+		frappe.qb.from_(Employee)
+		.select(
+			Employee.ach_last_used,
+			Employee.account_details_validated,
+			Employee.bank,
+			Employee.ach_account_type,
+			Employee.ach_prenote_date,
+			(Employee.employee_name).as_("party_name"),
+			(Employee.name).as_("party"),
+			ConstantColumn("Employee").as_("party_type"),
+		)
+		.where(Employee.bank.isnotnull())
+		.where(Coalesce(Employee.ach_last_used, "1900-1-1") <= getdate(filters.last_used_date))
+		.where(
+			Coalesce(Employee.account_details_validated, "1900-1-1") <= getdate(filters.validated_date)
+		)
+		.orderby(
+			Employee.ach_last_used,
+			Employee.account_details_validated,
+		)
+	)
+	if filters.ach_account_type == "Checking":
+		employee_query.where(Employee.ach_account_type == "Checking")
+	elif filters.ach_account_type == "Savings":
+		employee_query.where(Employee.ach_account_type == "Savings")
+	employees = employee_query.run(as_dict=True)
+
```

Comment:
For the ach_account_type filters here and in the Supplier Query block above, need to re-save that statement back to the query to add the where clause (if I change the filter to "Savings", it still shows the "Checking" rows):

`employee_query.where(Employee.ach_account_type == "Checking")` => `employee_query = employee_query.where(Employee.ach_account_type == "Checking")`

Reply:
Oh I see. I'll fix it